Automattic/wp-calypso
# Added pending post subscription list

Fixes https://github.com/Automattic/wp-calypso/issues/75966

## Proposed Changes

This PR adds the list of pending post subscriptions to the Subscription Manager:
<img width="935" alt="Screenshot 2023-04-19 at 19 50 31" src="https://user-images.githubusercontent.com/3832570/233179451-a5ff1bff-8b41-4cc6-8e82-3c7980755184.png">

This is the list of posts the user has initiated the subscription but hasn't confirmed yet. 

## Testing Instructions

1. Apply this PR and run the application.
2. Apply this patch to your sandbox
3. Subscribe to a blog with a non-WP.com user (for example, your-email+testing@gmail.com)
4. You should receive a confirmation e-mail that contains a "Manage subscriptions" button
5. Copy the button's link & visit in an incognito window
6. Open dev tools & copy the cookie. Make sure "Show URL decoded" is checked. Copy the value from the subkey cookie.
7. Add the subkey cookie to `calypso.localhost:3000`.
8. Go to `/subscriptions/pending`.
9. You should see the list of pending post subscriptions like in the screenshot above.
